src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_validators.py

```diff
@@ -0,0 +1,21 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def storage_account_id(cmd, namespace):
+    """Validate storage account name"""
+    from azure.cli.core.profiles import ResourceType
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+    if (namespace.storage_account and not
+            ('/providers/Microsoft.ClassicStorage/storageAccounts/' in namespace.storage_account or
+             '/providers/Microsoft.Storage/storageAccounts/' in namespace.storage_account)):
+        storage_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_STORAGE)
+        acc = storage_client.storage_accounts.get_properties(namespace.resource_group_name,
+                                                             namespace.storage_account)
+        if not acc:
+            raise ValueError("Storage account named '{}' not found in the resource group '{}'.".
+                             format(namespace.storage_account, namespace.resource_group_name))
+        namespace.storage_account = acc.id  # pylint: disable=no-member
```

Comment:
See this logic for the recommended way of implementing "name or ID" behavior:
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters

If it is NOT a valid ID (which there is a standard call for) then build the ID (in this case, assuming regular storageAccount not classic storage). Always *assume it exists* and let ARM fail if it doesn't. Don't do a GET just to make sure the account exists.  If it does exists, the check wasn't necessary and you made extra REST calls for no reason. If it doesn't exist, it will fail either way.

Reply:
Done